File: composeApp/src/commonMain/kotlin/com/hhp227/concafe/core/util/RatingUtils.kt
package com.hhp227.concafe.core.util

import kotlin.math.absoluteValue
import kotlin.math.roundToInt

object RatingUtils {
    fun formatOneDecimal(rating: Double): String {
        val rounded = (rating * 10).roundToInt()
        return formatScaledOneDecimal(rounded)
    }

    fun formatOneDecimal(rating: Float): String {
        return formatOneDecimal(rating.toDouble())
    }

    fun formatOneDecimalTruncated(rating: Double): String {
        val truncated = (rating * 10).toInt()
        return formatScaledOneDecimal(truncated)
    }

    private fun formatScaledOneDecimal(scaled: Int): String {
        val integer = scaled / 10
        val decimal = scaled.absoluteValue % 10
        return "$integer.$decimal"
    }
}

File: iosApp/iosApp/Core/Util/RatingUtils.swift
//
//  RatingUtils.swift
//  ConCafe
//
//  Created by 홍희표 on 4/19/26.
//

import Foundation

enum RatingUtils {
    private static let posixLocale = Locale(identifier: "en_US_POSIX")

    static func formatOneDecimal(_ rating: Double) -> String {
        String(format: "%.1f", locale: posixLocale, rating)
    }

    static func formatOneDecimal(_ rating: Float) -> String {
        formatOneDecimal(Double(rating))
    }

    static func formatOneDecimalTruncated(_ rating: Double) -> String {
        let truncated = (rating * 10).rounded(.towardZero) / 10
        return formatOneDecimal(truncated)
    }
}
